x/programs/rust/wasmlanche-sdk/src/program.rs

```diff
@@ -171,3 +195,38 @@ impl<K> BorshSerialize for CallProgramArgs<'_, K> {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{DeferDeserialize, ExternalCallError};
+
+    #[test]
+    fn defer_ok_variant() {
+        let bytes = DeferDeserialize(vec![1, 0]);
+        assert_eq!(bytes.deserialize::<Result<_, ()>>().unwrap(), Ok(false));
+    }
```

Comment:
This is interesting. What are you actually testing here? The implementation of the `derserialize` function is trivial, so I think you're actually testing the `BorshDeserialize` implementation of `Result<bool, ()>`.

Reply:
Removed it.